renderer/src/pages/Dashboard.tsx

```diff
@@ -42,16 +44,31 @@ const Dashboard = (): JSX.Element => {
     const unsubscribeOnActivityLogged = window.electron.stationEvents.onActivityLogged(setActivities)
     const unsubscribeOnEarningsChanged = window.electron.stationEvents.onEarningsChanged(setTotalEarnigs)
     const unsubscribeOnJobProcessed = window.electron.stationEvents.onJobProcessed(setTotalJobs)
+    const unsubscribeInAppNotification = window.electron.stationEvents.onNotification(setNotification)
 
     return () => {
       unsubscribeOnActivityLogged()
       unsubscribeOnEarningsChanged()
       unsubscribeOnJobProcessed()
+      unsubscribeInAppNotification()
     }
   }, [])
 
+  const notificationCallback = async () => {
+    if (notification) {
+      await notification.action()
+      setNotification(undefined)
+    }
+  }
+
   return (
     <div className="h-screen w-screen overflow-hidden bg-grayscale-100">
+      {notification &&
+        <Notification message={notification.text}
+        buttonText={notification.buttonText}
+        callback={notificationCallback} />
```

Comment:
`callback` isn't so semantic, what would be a better prop here? It's not `onDismiss`, it's more like `onClick` however it only fires when you click the button, not the whole notification.

I think we're generalizing this component too much, we don't yet have a concept of in-app notifications, can we refactor it to be specific to this use case? Something like `<UpdateBanner onRestart={}>`. Wdyt?

Reply:
Seems legit to me.
Hence, should the click `action`  be specific for this component? Calling directly a possible restart API instead of using the `action` property of the object?

Reply:
Good idea, the component can issue the rpc call itself 👍